packages/browser/src/index.test.ts

```diff
@@ -1,17 +1,50 @@
-import { createRequester } from '@logto/js';
+import { createRequester, generateCodeChallenge, parseUriParameters, QueryKey } from '@logto/js';
 
-import LogtoClient from './index';
+import LogtoClient, { getLogtoKey, LogtoSignInSessionItem } from './index';
+
+const clientId = 'client_id_value';
+const endpoint = 'https://logto.dev';
+const authorizationEndpoint = `${endpoint}/oidc/auth`;
+const requester = createRequester();
+
+jest.mock('@logto/js', () => ({
+  ...jest.requireActual('@logto/js'),
+  fetchOidConfig: async () => ({
+    authorization_endpoint: authorizationEndpoint,
+  }),
+}));
 
 describe('LogtoClient', () => {
   test('constructor', () => {
-    expect(
-      () =>
-        new LogtoClient({
-          endpoint: 'https://logto.dev',
-          clientId: 'client_id_value',
-          usingPersistStorage: false,
-          requester: createRequester(),
-        })
-    ).not.toThrow();
+    expect(() => new LogtoClient({ endpoint, clientId, requester })).not.toThrow();
+  });
+
+  describe('signInWithBrowser', () => {
+    test('sessionStorage and signInUri should contain correct properties', async () => {
+      const logtoClient = new LogtoClient({ endpoint, clientId, requester });
+      const originalRedirectUri = 'http://localhost:3000/callback';
+      await logtoClient.signInWithBrowser(originalRedirectUri);
+
+      const item = sessionStorage.getItem(getLogtoKey(clientId)) ?? '{}';
+      const logtoSignInSessionItem = JSON.parse(item) as LogtoSignInSessionItem;
```

Comment:
This is wrong, as the item could be `{}`, so the parsed object `logtoSignInSessionItem` is likely to be an empty object `{}`, which doesn't match the type `LogtoSignInSessionItem`

Reply:
I use `?? '{}'` and `as LogtoSignInSessionItem` on purpose.

If the item is null, `?? '{}'` will make logtoSignInSessionItem an empty object and  won't fail immediately.
And then I use the following `expect` statements (which can give a better message for the failed test case) to check whether sessionStorage contains the expected item with the expected properties.

`as LogtoSignInSessionItem` makes (e.g.) `logtoSignInSessionItem?.state` access legally (passing eslint).

Need some suggestions to help achieve these purposes for better UTs. @demonzoo

Reply:
Improved, @demonzoo PTAL